src/test/ui/issues/issue-23189/issue-23189.rs

```diff
@@ -0,0 +1,17 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+mod auxiliary {
+    mod module;
+}
+
+fn main() {
+    let _ = module { x: 0 }; //~ERROR cannot find struct
```

Comment:
After looking at the original issue, I'm wondering if the ICE only triggered when the name `module` was in scope. Putting it under `auxiliary` would change that.

Did you manage to reproduce the original ICE with this test?